server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -287,6 +288,8 @@ Runnable getGlobalCheckpointSyncer() {
     private volatile boolean useRetentionLeasesInPeerRecovery;
     private final boolean isDataStreamIndex; // if a shard is a part of data stream
 
+    private final ShardIndexingTimeStats shardIndexingTimeStats;
```

Comment:
Unfortunately, an index can be added and removed to/from a data stream using the `_data_stream/_modify` API. Let us leave that for a follow-up, but perhaps add a comment.

We could also just always collect this?

Reply:
That API only allows adding a backing (read) index to the data stream, right? In that case I think it won't be necessary to take those into account, even though it can skew the results a bit.